HaxeManual/10-std.tex

```diff
@@ -597,6 +597,170 @@ \subsection{Serialization format}
+
+\paragraph{Creating a root element}
+
+A \type{Xml} root element can be created using the \expr{Xml.createElement} method.
+\begin{lstlisting}
+var root = Xml.createElement('root');
+trace(root); // <root />
+\end{lstlisting}
+
+An root node element can also be created by parsing a \type{String} containing the XML data.
+\begin{lstlisting}
+var root = Xml.parse('<root />').firstElement();
+trace(root); // <root />
+\end{lstlisting}
+
+\paragraph{Creating child elements}
+
+Adding child elements to the root can be done using the \expr{createElement} method.
+\begin{lstlisting}
+var child:Xml = Xml.createElement('child'); 
+root.addChild(child);
+trace(root); // <root><child/></root>
+\end{lstlisting}
+
+
+Adding attributes to an element can be done by using the \expr{set()} method.
+\begin{lstlisting}
+child.set('name', 'John');
+trace(root); // <root><child name="John"/></root>
+\end{lstlisting}
+
+\paragraph{Accessing elements and values}
+
+This code parses an XML string into an object structure \type{Xml} and then accesses properties of the object.
+\begin{lstlisting}
+var xmlString = '<hello name="world!">Haxe is great!</hello>';
+var xml:Xml = Xml.parse(xmlString).firstElement();
+	
+trace(xml.nodeName); // hello
+trace(xml.get('name')); // world!
+trace(xml.firstChild().nodeValue); // Haxe is great!
+\end{lstlisting}
+
+The difference between \expr{firstChild} and \expr{firstElement} is that the second function will return the first child with the type \type{Xml.Element}. 
+
+\paragraph{Iterate on Xml elements}
+We can as well use other methods to iterate either over children or elements.
+\begin{lstlisting}
+for (child in xml) {
+	// iterate on all childs.
+}
+for (elt in xml.elements()) {
+	// iterate on all elements.
+}
+for (user in xml.elementsNamed("user")) {
+	// iterate on all elements with a nodeName "user".
+}
+for (att in xml.attributes()) {
+	// iterator on all attributes.
+}
+\end{lstlisting}
+
+\subsection{Parsing Xml}
+\label{std-Xml-parsing}
+
+Use the \expr{Xml.parse} static method to parse \emph{XML} data and obtain a Haxe value from it:
+
+\begin{lstlisting}
+var xml = Xml.parse('<root>Haxe is great!</root>').firstElement();
+trace(xml.firstChild().nodeValue);
+\end{lstlisting}
+
+\subsection{Encoding Xml}
+\label{std-Xml-encoding}
+
+Use the \expr{xml.toString()} method to encode a Haxe value into a \emph{XML} string:
+\begin{lstlisting}
+var xml = Xml.createElement('root');
+xml.addChild(Xml.createElement('child1'));
+xml.addChild(Xml.createElement('child2'));
+
+trace(xml.toString()); // <root><child1/><child2/></root>
+\end{lstlisting}
+
+\subsection{Fast Xml access}
+\label{std-Xml-Fast}
+
+While the base \type{Xml} API is easy to use, it is sometimes a bit painful to do all the checks of \type{Xml} correctness and can result in verbose code. The \href{http://api.haxe.org/haxe/xml/Fast.html}{haxe.xml.Fast} API is here to help by providing a fast dot-syntax access to the most common \type{Xml} methods.
```

Comment:
Can we use something other than "painful"? That's not something that should appear in a technical documentation in my opinion.

Maybe just lose the first sentence entirely?

Also "The API is here" sounds a bit silly. Would be better to go for "The API exists to ..."